rand_core/src/impls.rs

```diff
@@ -219,20 +219,38 @@ impl<R: BlockRngCore> BlockRng<R> {
     }
 
     /// Return a reference the wrapped `BlockRngCore`.
+    #[inline(always)]
     pub fn inner(&self) -> &R {
         &self.core
     }
 
     /// Return a mutable reference the wrapped `BlockRngCore`.
+    #[inline(always)]
     pub fn inner_mut(&mut self) -> &mut R {
         &mut self.core
     }
-
-    // Reset the number of available results.
-    // This will force a new set of results to be generated on next use.
+    
+    /// Get the index into the result buffer
+    #[inline(always)]
+    pub fn index(&self) -> usize {
+        self.index
+    }
+    
+    /// Reset the number of available results.
+    /// This will force a new set of results to be generated on next use.
+    #[inline(always)]
     pub fn reset(&mut self) {
         self.index = self.results.as_ref().len();
     }
+    
+    /// Generate a new set of results immediately, setting the index to the
+    /// given value.
```

Comment:
I see the use in combining both pieces of functionality, but doing it with a method named `generate` is a bit strange. I think it is better to just have `get_index`, `set_index` and `generate` separately.

Reply:
I did do that, but this provides an extra level of protection to avoid re-using the same results. It also seems to fit with the existing code extremely well.